### ISSUE TITLE
Wrong result for scalar-vector product

### ISSUE THREAD

**Issue by tkoolen** [JuliaAlgebra/MultivariatePolynomials.jl#104]
```julia
julia> using DynamicPolynomials

julia> @polyvar x
(x,)

julia> y = x * [x, x]
(PolyVar{true}[x, x])x

julia> typeof(y)
Term{true,Array{PolyVar{true},1}}
```

This calls https://github.com/JuliaAlgebra/MultivariatePolynomials.jl/blob/518eda684f81254234ddf558728af5ab18582a99/src/operators.jl#L27.

**Comment by blegat** [comment]
This is related to https://github.com/JuliaAlgebra/MultivariatePolynomials.jl/issues/103 and due to the design that what isn't a polynomial is a constant to avoid restricting constants to, say, `Number` so that constants can for example be matrices of `Float64` (we also never assume that `*` is commutative for this reason).
We should maybe make a special case for `AbstractArray` or polynomials and fallback to broadcast for operation such as `scalar * array`, `array * scalar` and `array / scalar`. Do you see any other ones ?

**Comment by saschatimme** [comment]
Fixed in #105.

**Comment by tkoolen** [comment]
Sorry for not responding to your earlier comment @blegat. Thanks @saschatimme!
